onnxscript/rewriter/_rewrite_rule.py

```diff
@@ -174,7 +174,14 @@ def try_rewrite(
                 if var.name is not None:
                     if var.name not in match.bindings:
                         match.bind(var.name, None)
-            check_match_result = self._condition_function(context, **match.bindings)
+            try:
+                check_match_result = self._condition_function(context, **match.bindings)
+            except _basics.MatchFailureError as e:
```

Comment:
I imagine the error is always caught here and not exposed to users, is that accurate?

Reply:
Yes